packages/redux/src/sharedWishlists/middlewares/updateSharedWishlistUponItemsChanges.ts

```diff
@@ -0,0 +1,45 @@
+import * as actionTypes from '../../wishlists/actionTypes.js';
+import {
+  getSharedWishlistId,
+  type UpdateSharedWishlistAction,
+} from '../index.js';
+import { getUser, getUserIsGuest } from '../../users/selectors.js';
+import { updateSharedWishlist } from '../actions/index.js';
+import type { AnyAction, Middleware } from 'redux';
+import type { GetOptionsArgument, StoreState } from '../../index.js';
+import type { ThunkDispatch } from 'redux-thunk';
+
+type UpdateSharedWishlistUponItemsChangesParams = {
+  // Redux action dispatch.
+  dispatch: ThunkDispatch<
+    StoreState,
+    GetOptionsArgument,
+    UpdateSharedWishlistAction
+  >;
+  // Returns the current redux state.
+  getState: () => StoreState;
+};
+
+const updateSharedWishlistUponItemsChanges: Middleware =
+  ({ dispatch, getState }: UpdateSharedWishlistUponItemsChangesParams) =>
+  next =>
+  (action: AnyAction) => {
+    if (
+      action.type === actionTypes.UPDATE_WISHLIST_ITEM_SUCCESS ||
+      action.type === actionTypes.ADD_WISHLIST_ITEM_SUCCESS ||
+      action.type === actionTypes.REMOVE_WISHLIST_ITEM_SUCCESS
+    ) {
+      const user = getUser(getState());
```

Comment:
What do you think of storing the `state` inside a const here? that way we would not have to call `getState()` function twice (here and on line 36).

Reply:
Agree! Updated